src/compiler/crystal/syntax/parser.cr

```diff
@@ -3266,7 +3259,7 @@ module Crystal
       @doc_enabled = !!@wants_doc
       pop_def
 
-      node = Def.new name, args, body, receiver, block_arg, return_type, (is_macro_def || @is_macro_def), @yields, is_abstract, splat_index, double_splat: double_splat, free_vars: free_vars
+      node = Def.new name, args, body, receiver, block_arg, return_type, @is_macro_def, @yields, is_abstract, splat_index, double_splat: double_splat, free_vars: free_vars
```

Comment:
Is `@is_macro_def` argument still needed?

Reply:
Yes, a `def` that mentions `@type` inside macro code is a macro def. So the concept exists and still remains, you just don't have to make it explicit with `macro def`.